src/test/java/core/basesyntax/CalculatorTest.java

```diff
@@ -0,0 +1,116 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class CalculatorTest {
+    private static Calculator calculator;
+
+    @BeforeAll
+    static void init() {
+        calculator = new Calculator();
+    }
+
+    @Test
+    public void addition_both_positive_ok() {
+        assertEquals(0, calculator.calculate(0, 0, '+'));
+        assertEquals(4, calculator.calculate(2, 2, '+'));
+        assertEquals(143234, calculator.calculate(72143, 71091, '+'));
+        assertEquals(4.294967294E9,
+                calculator.calculate(Integer.MAX_VALUE, Integer.MAX_VALUE, '+'));
+    }
+
+    @Test
+    public void addition_both_negative_ok() {
+        assertEquals(-17, calculator.calculate(-9, -8, '+'));
+        assertEquals(-143234, calculator.calculate(-72143, -71091, '+'));
+    }
+
+    @Test
+    public void addition_positive_negative_ok() {
+        assertEquals(-10, calculator.calculate(10, -20, '+'));
+        assertEquals(43, calculator.calculate(-21, 64, '+'));
+        assertEquals(0, calculator.calculate(Integer.MAX_VALUE, -Integer.MAX_VALUE, '+'));
+    }
+
+    @Test
+    public void subtraction_positive_ok() {
+        assertEquals(0, calculator.calculate(-0, -0, '+'));
+        assertEquals(1, calculator.calculate(3, 2, '-'));
+        assertEquals(72143, calculator.calculate(143234, 71091, '-'));
+        assertEquals(0, calculator.calculate(Integer.MAX_VALUE, Integer.MAX_VALUE, '-'));
+    }
+
+    @Test
+    public void subtraction_negative_result_ok() {
+        assertEquals(-17, calculator.calculate(-67, -50, '-'));
+        assertEquals(-1052, calculator.calculate(-72143, -71091, '-'));
+    }
+
+    @Test
+    public void subtraction_positive_result_ok() {
+        assertEquals(10, calculator.calculate(-20, -30, '-'));
```

Comment:
```suggestion
        assertEquals(50, calculator.calculate(20, -30, '-'));
```
let's check both positive and negative input